src/extensions/default/JavaScriptCodeHints/ScopeManager.js

```diff
@@ -1334,7 +1344,7 @@ define(function (require, exports, module) {
             // We don't reset if the debugging flag is set
             // because it's easier to debug if the module isn't
             // getting reset all the time.
-            if (force || (!config.noReset && ++_hintCount > MAX_HINTS)) {
+            if (currentModule.resetForced || force || (!config.noReset && ++_hintCount > MAX_HINTS)) {
```

Comment:
Should `currentModule.resetForced` be set back to `false` at some point? 🤔

Reply:
`currentModule` is reassigned with `newTernModule` soon after ...

Reply:
Thanks for confirming 👍